SteamKit2/SteamKit2/Steam/Handlers/SteamMatchmaking/Lobby.cs

```diff
@@ -0,0 +1,402 @@
+            /// </summary>
+            public ELobbyType LobbyType { get; }
+
+            /// <summary>
+            /// The lobby's flags.
+            /// </summary>
+            public int LobbyFlags { get; }
+
+            /// <summary>
+            /// The SteamID of the lobby's owner. Please keep in mind that Steam does not provide lobby
+            /// owner details for lobbies returned in a lobby list. As such, lobbies that have been
+            /// obtained/updated as a result of calling <see cref="SteamMatchmaking.GetLobbyList"/>
+            /// may have a null (or non-null but state) owner.
+            /// </summary>
+            public SteamID OwnerSteamID { get; }
+
+            /// <summary>
+            /// The metadata of the lobby; string key-value pairs.
+            /// </summary>
+            public IReadOnlyDictionary<string, string> Metadata { get; }
+
+            /// <summary>
+            /// The maximum number of members that can occupy the lobby.
+            /// </summary>
+            public int MaxMembers { get; }
+
+            /// <summary>
+            /// The number of members that are currently occupying the lobby.
+            /// </summary>
+            public int NumMembers { get; }
+
+            /// <summary>
+            /// A list of lobby members. This will only be populated for the user's current lobby.
+            /// </summary>
+            public IReadOnlyList<Member> Members { get; }
+
+            /// <summary>
+            /// The distance of the lobby.
+            /// </summary>
+            public float? Distance { get; }
+
+            /// <summary>
+            /// The weight of the lobby.
+            /// </summary>
+            public long? Weight { get; }
+
+            static readonly IReadOnlyDictionary<string, string> EmptyMetadata =
+                new ReadOnlyDictionary<string, string>( new Dictionary<string, string>() );
+
+            static readonly IReadOnlyList<Member> EmptyMembers = new List<Member>().AsReadOnly();
+
+            internal Lobby( SteamID steamId, ELobbyType lobbyType, int lobbyFlags, SteamID ownerSteamId, IReadOnlyDictionary<string, string> metadata,
+                int maxMembers, int numMembers, IReadOnlyList<Member> members, float? distance, long? weight )
+            {
+                SteamID = steamId;
+                LobbyType = lobbyType;
+                LobbyFlags = lobbyFlags;
+                OwnerSteamID = ownerSteamId;
+                Metadata = metadata ?? EmptyMetadata;
+                MaxMembers = maxMembers;
+                NumMembers = numMembers;
+                Members = members ?? EmptyMembers;
+                Distance = distance;
+                Weight = weight;
+            }
+
+            internal static byte[] EncodeMetadata( IReadOnlyDictionary<string, string> metadata )
+            {
+                if ( metadata == null )
+                {
+                    metadata = EmptyMetadata;
```

Comment:
Why do we assign EmptyMetadata here and then iterate over it? If we're doing a null check anyway, then we can just do `if (metadata != null) { foreach (var entry in metadata) { ... } }`.

Reply:
I don't think I thought about it _that_ hard when I wrote it. However, I'd say my thought process was something along the lines of "this is essentially a default parameter so we'll handle it up front and minimise cyclomatic" i.e. it's optimised for comprehension rather than runtime efficiency.

However, honestly, I really don't have a strong opinion about it.